[ERROR] FATAL UNHANDLED EXCEPTION: System.ArgumentException: An item with the same key has already been added.

Hi,

This is a fantastic and easy-to-use plugin, which as made my development life much easier!

I'm getting an exception thrown when I'm checking the `Location` permission:

    [ERROR] FATAL UNHANDLED EXCEPTION: System.ArgumentException: An item with the same key has already been added. Key: Location
        at System.Collections.Generic.Dictionary`2[TKey,TValue].TryInsert (TKey key, TValue value, System.Collections.Generic.InsertionBehavior behavior) [0x000bb] in <896ad1d315ca4ba7b117efb8dacaedcf>:0 
        at System.Collections.Generic.Dictionary`2[TKey,TValue].Add (TKey key, TValue value) [0x00000] in <896ad1d315ca4ba7b117efb8dacaedcf>:0 

This has started to appear more often recently (using v2.1.0 of the Permissions plugin). The code I am using for checking permissions is as follows:

    var result = await Plugin.Permissions.CrossPermissions.Current.RequestPermissionsAsync(new[] { Plugin.Permissions.Abstractions.Permission.Location });
    var status = result[Plugin.Permissions.Abstractions.Permission.Location];
    if (status != Plugin.Permissions.Abstractions.PermissionStatus.Granted)
    {
        // Display warning
    }

Because I am getting this exception thrown more often now, I figured it would be something to do with the plugin itself, as this wasn't happening 2-3 weeks ago. If not, what is wrong with how I check my permissions? 

Thanks

--- a/src/Plugin.Permissions.Android/PermissionsImplementation.cs
+++ b/src/Plugin.Permissions.Android/PermissionsImplementation.cs
@@ -130,10 +130,11 @@ namespace Plugin.Permissions
                 Debug.WriteLine("Unable to detect current Activity. Please ensure Plugin.CurrentActivity is installed in your Android project and your Application class is registering with Application.IActivityLifecycleCallbacks.");
                 foreach (var permission in permissions)
                 {
-                    if (results.ContainsKey(permission))
-                        continue;
-
-                    results.Add(permission, PermissionStatus.Unknown);
+					lock (locker)
+					{
+						if (!results.ContainsKey(permission))
+							results.Add(permission, PermissionStatus.Unknown);
+					}
                 }
 
                 return results;
@@ -151,7 +152,8 @@ namespace Plugin.Permissions
                     {
                         lock (locker)
                         {
-                            results.Add(permission, PermissionStatus.Unknown);
+							if (!results.ContainsKey(permission))
+								results.Add(permission, PermissionStatus.Unknown);
                         }
                         continue;
                     }
@@ -163,7 +165,8 @@ namespace Plugin.Permissions
                     //if we are granted you are good!
                     lock (locker)
                     {
-                        results.Add(permission, PermissionStatus.Granted);
+						if (!results.ContainsKey(permission))
+							results.Add(permission, PermissionStatus.Granted);
                     }
                 }
             }

